### PR TITLE
Feature/restructuring

### PR DESCRIPTION
**Pull request by jamclar**
Changes made to incorporate login/logout and displaying the current user at the top of the page

Changes also made to start refactoring app.py into a more readable format

The "collections", "contact" and "about" endpoints have been removed from app.py, and now sit in a folder called "routes".
They are then imported from routes into app.py, and the functionality remains the same.

There is also a folder named "authentication" which keeps all code relating to authentication.

### CODE REVIEW

**Comment by jamclar** [comment]
I am aware of other ongoing work which is currently being pushed to the Master branch, this will need to be checked before completing this MR

**Review by lukegorman — APPROVED** [review]
Superb!

File: nvsvocprez/view/templates/page.html [hunk 1]
```diff
@@ -18,17 +18,20 @@
       <a href="https://www.bodc.ac.uk/"><img src="/static/bodc-bw.png" alt="BODC logo" style="height:70%; float:right ;" /></a>
     </div>
     <div id="header-title" style="grid-row:2;">
-
-      {% if login_status == 'Not Logged in' %}
+    {% if logged_in_user != '' %}
+      {% if logged_in_user == 'Not Logged in' %}
```
Comment (lukegorman) [inline nvsvocprez/view/templates/page.html:22]: make note logged in string enum

File: nvsvocprez/view/templates/page.html [hunk 2]
```diff
@@ -18,17 +18,20 @@
...
       {% else %}
         <a href="/logout">Log-out</a>
       {% endif %}
+    {% endif %}
       <span>The NERC Vocabulary Server (NVS)</span>
       <a href="https://sdc.ui.argo.grnet.gr/sdc/dashboard/Critical/VOCABULARY">Service Status</a>
     </div>
-    <div >
+    <div>
+    {% if logged_in_user != '' %}
```
Comment (lukegorman) [inline nvsvocprez/view/templates/page.html:32]: initialise logged_in_code to another enum

File: nvsvocprez/utilities/config.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+"""Reads .env file and sends the config to authentication processes."""
+from starlette.config import Config
+
+
+class MissingKeyError(Exception):
+    """Raised when the env file is missing one or more keys."""
+
+
+expected_config_keys = {
```
Comment (lukegorman) [inline nvsvocprez/utilities/config.py:9]: login secrets are "REQUIRED" but only if login is enabled. Modify conditionals accordingly.

**Comment by lukegorman** [comment]
approved

File: nvsvocprez/routes/utils.py [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    Returns:
+        str: The construncted sparql query.
+    """
+
+    prefix_text = ""
+    filter_text = ""
+    if profile.id != "nvs":
+        # Build prefix text.
+        for ontology, data in profile.ontologies.items():
+            prefix_text += f'PREFIX {data["prefix"]}: <{data["url"]}#>\n'
```
Comment (lukegorman) [inline nvsvocprez/routes/utils.py:353]: woops

**Pull request merged**